docs/firestore/emulator.md

```diff
@@ -0,0 +1,50 @@
+---
+title: Using The Emulator With Firestore
+description: Using the emulator to test your app locally
+next: /firestore/usage-with-flatlists
+previous: /firestore/usage
+---
+
+You can test your app and its Firestore implementation with an emulator which is built to mimic the behavior of Cloud Firestore. This means you can connect your app directly to the emulator to perform integration testing or QA without touching production data.
+
+For example, you can connect your app to the emulator to safely read and write documents in testing.
+
+# Running the emulator
+
+To be able to mimic the behavior of Cloud Firestore, you need to run the emulator. This is effectively a server that will receive and send requests in lieu of Cloud Firestore. This is achieved by running the following commands:
+
+```bash
+// install the Firebase CLI which will run the emulator
+curl -sL firebase.tools | bash
+
+// run this command to start the emulator, it will also install it if this is your first time running the command
+firebase emulators:start --only firestore
+```
+
+You should see a `firestore-debug.log` file in the directory you ran the command which will have a log of all the requests.
+
+# Connect to emulator from your app
+
+You need to configure the following property as soon as possible in the startup of your application:
+
+```jsx
+import '@react-native-firebase/app';
+import firestore from '@react-native-firebase/firestore';
+
+const db = firestore();
+
+// set the host property to connect to the emulator
+// set these before any read/write operations occur to ensure it doesn't affect your Cloud Firestore data!
+
+db.settings({ host: 'http://localhost:8080' });
```

Comment:
Is with the `http` protocol part correct? Though it was without on the previous docs you had?

If it's without then this and js replacer code below would need changing.

Reply:
I tried to make it consistent with how we approached it in cloud functions emulator which passes along the `http`protocol. AlthoughI have changed what we pass under the hood to Android. If you pass along the `http` protocol, the app crashes.